examples/text_summarization/abstractive_summarization_minilm_cnndm.ipynb

```diff
@@ -0,0 +1,552 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#train_dataset = processor.s2s_dataset_from_json_or_file(\"/dadendev/unilm/data/xsum.train.uncased_tokenized.json\", train_mode=True, top_n=TOP_N)\n",
+    "#test_dataset = processor.s2s_dataset_from_json_or_file(\"/dadendev/unilm/data/xsum.test.uncased_tokenized.json\", train_mode=False, top_n=TOP_N)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#import torch\n",
+    "#torch.save(train_dataset, os.path.join(CACHE_DIR, \"cnndm_train_dataset.pt\"))\n",
+    "#torch.save(test_dataset, os.path.join(CACHE_DIR, \"cnndm_test_dataset.pt\"))\n",
+    "#train_dataset = torch.load(os.path.join(CACHE_DIR, \"train_dataset.pt\"))\n",
+    "#test_dataset = torch.load(os.path.join(CACHE_DIR, \"test_dataset.pt\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fine tune model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `S2SAbstractiveSummarizer` loads a pre-trained UniLM model specified by `model_name`.  \n",
+    "Call `S2SAbstractiveSummarizer.list_supported_models()` to see all the supported models.  \n",
+    "If you want to use a model on the local disk, specify `load_model_from_dir` and `model_file_name`. This is particularly useful if you want to load a previously fine-tuned model and use it for inference directly without fine-tuning. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "S2SAbstractiveSummarizer.list_supported_models()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "abs_summarizer = S2SAbstractiveSummarizer(\n",
+    "    model_name=MODEL_NAME,\n",
+    "    max_seq_length=MAX_SEQ_LENGTH,\n",
+    "    max_source_seq_length=MAX_SOURCE_SEQ_LENGTH,\n",
+    "    max_target_seq_length=MAX_TARGET_SEQ_LENGTH,\n",
+    "    cache_dir=CACHE_DIR\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "abs_summarizer.model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\"\"\"\n",
+    "abs_summarizer = S2SAbstractiveSummarizer(\n",
+    "     model_name=MODEL_NAME,\n",
+    "     max_seq_length=MAX_SEQ_LENGTH,\n",
+    "     max_source_seq_length=MAX_SOURCE_SEQ_LENGTH,\n",
+    "    max_target_seq_length=MAX_TARGET_SEQ_LENGTH,\n",
+    "     load_model_from_dir=RESULT_DIR,\n",
+    "    model_file_name=\"model.5000.bin\",\n",
+    " )\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#\"\"\"\n",
+    "abs_summarizer.fit(\n",
```

Comment:
Would be nice to include a Timer and display the training time in the main notebook to show the advantage of using minilm over other models.